### PR TITLE
Handle issues 7, 9, and 10.

### PR DESCRIPTION
**Pull request by KarlSchimpf**
Add section to describe the textual version of exception handling. Includes clarifying that try blocks are implicitly labeled. Fixes for issues #7 and #9.

Use 'catch default' in textual version and instruction 'else' in the binary format #9.

Disambiguate `rethrow` when in nested catch blocks (issue #10).

### CODE REVIEW

File: proposals/Exceptions.md
Comment (binji) [inline proposals/Exceptions.md]: Here's a cool trick: you can actually number 1. 1. 1. 1. and markdown will autonumber for you. :-)

File: proposals/Exceptions.md
Comment (binji) [inline proposals/Exceptions.md]: sp: `catch_default`

File: proposals/Exceptions.md
Comment (binji) [inline proposals/Exceptions.md]: Looks like the doc is inconsistent between `catch default` and `catch_default`

File: proposals/Exceptions.md
Comment (binji) [inline proposals/Exceptions.md]: sp: Zero

File: proposals/Exceptions.md
Comment (binji) [inline proposals/Exceptions.md]: I found this hard to understand until I saw @rossberg-chromium's description [here](https://github.com/WebAssembly/exception-handling/issues/10#issue-228061591). Include it below?

File: proposals/Exceptions.md
Comment (binji) [inline proposals/Exceptions.md]: exceptidx

File: proposals/Exceptions.md
Comment (binji) [inline proposals/Exceptions.md]: rethrow catch_idx

File: proposals/Exceptions.md
Comment (KarlSchimpf) [reply proposals/Exceptions.md]: Nice to know! Changing...

File: proposals/Exceptions.md
Comment (KarlSchimpf) [reply proposals/Exceptions.md]: Fixing to always use `catch default`

File: proposals/Exceptions.md
Comment (KarlSchimpf) [reply proposals/Exceptions.md]: Fixed.

File: proposals/Exceptions.md
Comment (KarlSchimpf) [reply proposals/Exceptions.md]: Fixed.

File: proposals/Exceptions.md
Comment (KarlSchimpf) [reply proposals/Exceptions.md]: Fixed

File: proposals/Exceptions.md
Comment (KarlSchimpf) [reply proposals/Exceptions.md]: Fixed.

File: proposals/Exceptions.md
Comment (KarlSchimpf) [reply proposals/Exceptions.md]: Added an example.

File: proposals/Exceptions.md
Comment (rossberg) [reply proposals/Exceptions.md]: Bikeshed: can we name it `catch_all`?

File: proposals/Exceptions.md
Comment (rossberg) [inline proposals/Exceptions.md]: We could model this as an ordinary label, which must be the label of a `try` instruction. Then no new concept is needed.

File: proposals/Exceptions.md
Comment (rossberg) [inline proposals/Exceptions.md]: Note that this points to the abstract syntax, not the text format. The latter will be under https://github.com/WebAssembly/spec/blob/master/document/text/instructions.rst, once spec PR #471 has landed.

File: proposals/Exceptions.md
Comment (KarlSchimpf) [reply proposals/Exceptions.md]: Changing to catch_all.

File: proposals/Exceptions.md
Comment (KarlSchimpf) [reply proposals/Exceptions.md]: I agree that not introducing a new concept is good. Changing.

File: proposals/Exceptions.md
Comment (KarlSchimpf) [reply proposals/Exceptions.md]: Changed link.

**Comment by KarlSchimpf** [comment]
Can I get a LGTM for this PR?

File: proposals/Exceptions.md
Comment (rossberg) [inline proposals/Exceptions.md]: I suppose this is meant to be `block`?

File: proposals/Exceptions.md
Comment (rossberg) [inline proposals/Exceptions.md]: Could add that using a label that does not reference a `try` instruction is a validation error.

**Review by rossberg — APPROVED** [review]
LGTM modulo nits.

File: proposals/Exceptions.md
Comment (backes) [inline proposals/Exceptions.md]: Two "example" in one sentence.

File: proposals/Exceptions.md
Comment (backes) [inline proposals/Exceptions.md]: Maybe change this to "caught by an enclosing catch block" here. Which exactly is detailed below.

File: proposals/Exceptions.md
Comment (backes) [inline proposals/Exceptions.md]: Capitalization: "This".

File: proposals/Exceptions.md
Comment (backes) [inline proposals/Exceptions.md]: Instructions are sometimes in italics (*throw*), sometimes set as code (`throw`). You might want to harmonize that.

File: proposals/Exceptions.md
Comment (backes) [inline proposals/Exceptions.md]: Typo: relativew_depth

File: proposals/Exceptions.md
Comment (backes) [inline proposals/Exceptions.md]: Double space before "and try".

File: proposals/Exceptions.md
Comment (backes) [inline proposals/Exceptions.md]: Trailing newline.

File: proposals/Exceptions.md
Comment (KarlSchimpf) [reply proposals/Exceptions.md]: Good point. Fixing.

File: proposals/Exceptions.md
Comment (KarlSchimpf) [reply proposals/Exceptions.md]: Added comment describing why `rethrow 2` is not allowed because it doesn't reference a try block. Also fixed `rethow 4` which should be `rethrow 3`.

File: proposals/Exceptions.md
Comment (KarlSchimpf) [inline proposals/Exceptions.md]: Fixed this to be `rethrow label`

File: proposals/Exceptions.md
Comment (KarlSchimpf) [inline proposals/Exceptions.md]: Fixed this sentence to refer to a label, not a catch_index.

File: proposals/Exceptions.md
Comment (KarlSchimpf) [reply proposals/Exceptions.md]: Done.

File: proposals/Exceptions.md
Comment (KarlSchimpf) [reply proposals/Exceptions.md]: Fixed.

File: proposals/Exceptions.md
Comment (KarlSchimpf) [inline proposals/Exceptions.md:220]: Added sentence to clarify that `rethrow 2` is not allowed.

File: proposals/Exceptions.md
Comment (KarlSchimpf) [reply proposals/Exceptions.md]: Done.

File: proposals/Exceptions.md
Comment (KarlSchimpf) [reply proposals/Exceptions.md]: Good point. Reconciling.

File: proposals/Exceptions.md
Comment (KarlSchimpf) [reply proposals/Exceptions.md]: Fixed.

File: proposals/Exceptions.md
Comment (KarlSchimpf) [reply proposals/Exceptions.md]: Fixed.

File: proposals/Exceptions.md
Comment (KarlSchimpf) [reply proposals/Exceptions.md]: Removed.

**Review by KarlSchimpf — COMMENTED** [review]
Fixed issues raised.

**Pull request merged**
